src/DynamicDiscreteSamplers.jl

```diff
@@ -807,7 +812,15 @@ end
 # Conform to the AbstractArray API
 Base.size(w::Weights) = (w.m[1],)
 
-# Define convinience constructors TODO: these can be significantly optimized, especially when `x isa Weights`
+for T in [:FixedSizeWeights, :SemiResizableWeights, :ResizableWeights]
+    @eval function $T(x::Weights)
+        m = Memory{UInt64}(undef, length(x.m))
+        unsafe_copyto!(m, 1, x.m, 1, length(x.m))
+        $(Symbol(:_, T))(m)
+    end
+end
```

Comment:
This seems like a simpler and equivalent approach:
```suggestion
FixedSizeWeights(x::Weights) = _FixedSizeWeights(copy(x.m))
SemiResizableWeights(x::Weights) = _SemiResizableWeights(copy(x.m))
ResizableWeights(x::Weights) = _ResizableWeights(copy(x.m))
```

Reply:
yes, I probably over-engineered this :D